addon/adapters/preprint.js

```diff
@@ -2,15 +2,25 @@ import OsfAdapter from './osf-adapter';
 
 export default OsfAdapter.extend({
     // Overrides updateRecord on OsfAdapter. Is identical to JSONAPIAdapter > Update Record (parent's parent method).
-    // Updates to preprints do not need special handling.
     updateRecord(store, type, snapshot) {
-        var data = {};
-        var serializer = store.serializerFor(type.modelName);
+        let data = {};
+        let url = null;
 
-        serializer.serializeIntoHash(data, type, snapshot, { includeId: true });
-
-        var id = snapshot.id;
-        var url = this.buildURL(type.modelName, id, snapshot, 'updateRecord');
+        if (snapshot.record.get('_dirtyRelationships')['node'] && snapshot.record.get('_dirtyRelationships')['node']['remove'].length && !snapshot.record.get('_dirtyRelationships')['node']['add'].length) {
```

Comment:
It seems with the current setup, `updateRecord()` can remove a node relationship of a preprint, or it can update a preprint's other attributes and non-node relationships, but not both at the same time. Does that have any side effects for how we use `.save()` on a `preprint` instance?

Reply:
In my opinion, this is only an issue if we need to update a preprint attribute and remove a node simultaneously, which we do not.  (This is also new functionality; it was previously impossible to remove the node relationship from the preprint). You can still *update a preprint attribute and add/update a node relationship* simultaneously, as before.  However, to unset a node, this has to be a request to a different endpoint. 

In `ember-osf-preprints`, there is one location where we can remove the node from the preprint.  It is unset locally and then immediately persisted to the server.

Reply:
That's fine. We just need to keep in mind that we cannot remove a node relationship and update other attributes at the same time. Should we add such a notice somewhere to make sure other developers won't run into this problem in the future if they ever try to do this?

Reply:
yes, good idea.